lib/chef/knife/ssh.rb

```diff
@@ -219,23 +219,28 @@ def search_nodes
       # @return [Hash<Symbol, Object>]
       def session_options(host, port, user=nil)
         ssh_config = Net::SSH.configuration_for(host)
-        {}.tap do |opts|
+        ssh_config.tap do |opts|
```

Comment:
`Net::SSH.configuration_for` returns several useful options, such as compression_level, timeout and etc.

But, session_options discards them.